### PR TITLE
fix(ui): fix sorting of parameters. Make the Remove override button clickable again

### PR DESCRIPTION
**Pull request by alexef**
- sorting by override was overridden by sorting by name
 
Screenshot 2022-11-16 at 15 33 39 (overrides will now show always on top of the list)

- button "Remove Override" had z-index:1 which is behind input z-index: 10

Screenshot 2022-11-16 at 15 33 02

Signed-off-by: Alex Eftimie <alex.eftimie@getyourguide.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: ui/src/app/applications/components/application-parameters/application-parameters.tsx [hunk 1]
```diff
@@ -192,29 +190,31 @@ export const ApplicationParameters = (props: {
...
-                            <label>values.yaml</label>
-                            <Expandable>
-                                <pre>{props.details.helm.values}</pre>
-                            </Expandable>
-                        </div>
-                    )}
-                </div>
-            )
-        });
+        if (app.spec.source.helm.values) {
```
Comment (rbreeze) [inline ui/src/app/applications/components/application-parameters/application-parameters.tsx:193]: Can you add `?`'s here to fail gracefully?

`app?.spec?.source?.helm?.values`

**Review by rbreeze — CHANGES_REQUESTED** [review]

File: ui/src/app/applications/components/application-parameters/application-parameters.tsx (hunk 1 above)
Comment (alexef) [reply ui/src/app/applications/components/application-parameters/application-parameters.tsx]: this is a bit excessive, at least app.spec.source should always be defined. but it's better to be safe, than sorry :)

**Review by rbreeze — APPROVED** [review]

**Comment by alexef** [comment]
@crenshaw-dev should we cherry pick this to 2.5? see #11506

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.5 for 2.5.4.

**Pull request merged**